### PR TITLE
Ce/signup analytics

### PR DESCRIPTION
**Pull request by calellowitz**
@sravfeyn @biyeun @czue 
This is a replacement PR for https://github.com/dimagi/commcare-hq/pull/12780
I branched off https://github.com/dimagi/commcare-hq/pull/12782 to incorporate the removal of the a/b test, and so if we merge this after (or instead of) that one, all of the signup analytics will be working again.

### CODE REVIEW

**Comment by biyeun** [comment]
rebased master, kicking tests

**Comment by biyeun** [comment]
@calellowitz this seems fine. I'm going to go ahead and close that previous PR

File: corehq/apps/registration/urls.py [hunk 1]
```diff
@@ -2,17 +2,16 @@
...
-    url(r'^process/?$', ProcessRegistrationView.as_view(), name=ProcessRegistrationView.urlname),
-    url(r'^user/?$', 'register_user', name='register_user'),
-    url(r'^domain/$', RegisterDomainView.as_view(), name='registration_domain'),
-    url(r'^domain/confirm(?:/(?P<guid>\w+))?/$', 'confirm_domain', name='registration_confirm_domain'),
-    url(r'^resend/$', 'resend_confirmation', name='registration_resend_domain_confirmation'),
-    url(r'^eula_agreement/$', 'eula_agreement', name="agree_to_eula"),
-)
+                       url(r'^$', 'registration_default', name='registration_default'),
+                       url(r'^user/new/?$', UserRegistrationView.as_view(), name=UserRegistrationView.urlname),
+                       url(r'^process/?$', ProcessRegistrationView.as_view(), name=ProcessRegistrationView.urlname),
```
Comment (dimagimon) [inline corehq/apps/registration/urls.py]: E501 line too long (116 > 115 characters)

File: corehq/apps/registration/urls.py [hunk 2]
```diff
@@ -2,17 +2,16 @@
...
+                       url(r'^domain/$', RegisterDomainView.as_view(), name='registration_domain'),
+                       url(r'^domain/confirm(?:/(?P<guid>\w+))?/$', 'confirm_domain', name='registration_confirm_domain'),
```
Comment (dimagimon) [inline corehq/apps/registration/urls.py]: E501 line too long (122 > 115 characters)

File: corehq/apps/registration/urls.py [hunk 3]
```diff
@@ -2,17 +2,16 @@
...
-    url(r'^process/?$', ProcessRegistrationView.as_view(), name=ProcessRegistrationView.urlname),
-    url(r'^user/?$', 'register_user', name='register_user'),
-    url(r'^domain/$', RegisterDomainView.as_view(), name='registration_domain'),
-    url(r'^domain/confirm(?:/(?P<guid>\w+))?/$', 'confirm_domain', name='registration_confirm_domain'),
-    url(r'^resend/$', 'resend_confirmation', name='registration_resend_domain_confirmation'),
-    url(r'^eula_agreement/$', 'eula_agreement', name="agree_to_eula"),
-)
+                       url(r'^$', 'registration_default', name='registration_default'),
+                       url(r'^user/?$', UserRegistrationView.as_view(), name=UserRegistrationView.urlname),
+                       url(r'^process/?$', ProcessRegistrationView.as_view(), name=ProcessRegistrationView.urlname),
```
Comment (dimagimon) [inline corehq/apps/registration/urls.py]: E501 line too long (116 > 115 characters)

File: corehq/apps/registration/urls.py [hunk 4]
```diff
@@ -2,17 +2,16 @@
...
+                       url(r'^domain/$', RegisterDomainView.as_view(), name='registration_domain'),
+                       url(r'^domain/confirm(?:/(?P<guid>\w+))?/$', 'confirm_domain', name='registration_confirm_domain'),
```
Comment (dimagimon) [inline corehq/apps/registration/urls.py]: E501 line too long (122 > 115 characters)

**Comment by orangejenny** [comment]
Test failure looks possibly real
```
FAIL: test_autocomplete_disabled (corehq.apps.registration.tests.test_views.TestNewWebUserRegistrationFormAutocomplete)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/registration/tests/test_views.py", line 12, in test_autocomplete_disabled
    self.verify(False, reverse("register_user"), "full_name", "email")
  File "/mnt/commcare-hq-ro/corehq/apps/domain/tests/test_views.py", line 151, in verify
    self.assertEqual(is_enabled, autocomplete_enabled)
AssertionError: True != False
```

**Comment by biyeun** [comment]
@orangejenny that test seems no longer relevant for this UI. @millerdev i'm deleting it---shows you as the original author. What functionality was this testing exactly?

**Comment by calellowitz** [comment]
:white_check_mark:

**Comment by calellowitz** [comment]
:white_check_mark:

**Comment by biyeun** [comment]
@orangejenny @benrudolph 😻

File: corehq/apps/registration/tests/test_views.py [hunk 5]
```diff
@@ -1,12 +0,0 @@
...
-from corehq.apps.domain.tests.test_views import BaseAutocompleteTest
-
-
-class TestNewWebUserRegistrationFormAutocomplete(BaseAutocompleteTest):
-
-    def test_autocomplete_enabled(self):
-        self.verify(True, reverse("register_user"), "full_name", "email")
-
-    def test_autocomplete_disabled(self):
-        self.verify(False, reverse("register_user"), "full_name", "email")
```
Comment (millerdev) [inline corehq/apps/registration/tests/test_views.py]: re: https://github.com/dimagi/commcare-hq/pull/12796#issuecomment-240134603 by @biyeun 

Autocomplete of some fields on the signup page needs to be disabled for ICDS per requirements of the security audit. This is (was?) configured with `ENABLE_DRACONIAN_SECURITY_FEATURES` setting. The [test failure](https://github.com/dimagi/commcare-hq/pull/12796#issuecomment-240104659) seems like a real failure, and I think this test is not irrelevant.

**Comment by biyeun** [comment]
@millerdev was this browser autocomplete? The structure of the signup form has changed quite dramatically since the revamp, and the username and password fields are not autocompleted by default

**Comment by millerdev** [comment]
yes, browser autocomplete. username and password should not autocomplete for sure, but this test was checking that other fields like *full name* and *email* were not autocompleted. If they're not then it seems fine to delete those tests, although I think the test failure was saying autocomplete is enabled (not disabled as it should be) for those fields when `ENABLE_DRACONIAN_SECURITY_FEATURES=True`.

**Pull request merged**